### PR TITLE
Bug 2100436: VM list filter failed vms

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Add `Failed` status filter

## 🎥 Demo

Screenshot from 2022-07-01 17-05-30
Screenshot from 2022-07-01 17-05-20
Screenshot from 2022-07-01 17-05-13

### CODE REVIEW

File: src/views/virtualmachines/utils/virtualMachineRowFilter.ts [hunk 1]
```diff
@@ -13,21 +13,26 @@ import {
...
-        !statuses?.all?.find((s) => s === status)
+        id === obj?.status?.printableStatus ||
+        (id === printableVMStatus.Failed && isFailedPrintableStatus(obj?.status?.printableStatus))
       );
     },
+    filter: (statuses, obj) => {
+      const status = obj?.status?.printableStatus;
+
+      if (isFailedPrintableStatus(status) && statuses.selected?.includes(printableVMStatus.Failed))
+        return true;
```
[the remark this answers is not in the document]
Comment (upalatucci) [reply src/views/virtualmachines/utils/virtualMachineRowFilter.ts:32]: Okay great. But it's a little too long and hard to comprehend like this in my opinion. I'll create a variable for the failed status part

File: src/views/virtualmachines/utils/virtualMachineRowFilter.ts [hunk 2]
```diff
@@ -13,21 +13,26 @@ import {
...
+import { isFailedPrintableStatus, printableVMStatus } from './virtualMachineStatuses';
 
 export const getStatusFilter = (t: TFunction): RowFilter[] => [
   {
     filterGroupName: t('Status'),
     type: 'status',
-    reducer: (obj) => obj?.status?.printableStatus,
-    filter: (statuses, obj) => {
-      const status = obj?.status?.printableStatus;
+    isMatch: (obj, id) => {
```
[the remark this answers is not in the document]
Comment (avivtur) [reply src/views/virtualmachines/utils/virtualMachineRowFilter.ts]: I think I would go with `selectedStatus` :)

**Pull request merged**